source/backend/dal/Repositories/CompensationRequisitionRepository.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Pims.Dal.Entities;
+using Pims.Dal.Helpers.Extensions;
+using Pims.Dal.Security;
+
+namespace Pims.Dal.Repositories
+{
+    /// <summary>
+    /// Provides a repository to interact with compensation requisitions within the datasource.
+    /// </summary>
+    public class CompensationRequisitionRepository : BaseRepository<PimsCompensationRequisition>, ICompensationRequisitionRepository
+    {
+        #region Constructors
+
+        /// <summary>
+        /// Creates a new instance of a CompensationRepository, and initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="dbContext"></param>
+        /// <param name="user"></param>
+        /// <param name="logger"></param>
+        public CompensationRequisitionRepository(PimsContext dbContext, ClaimsPrincipal user, ILogger<BaseRepository> logger)
+            : base(dbContext, user, logger)
+        {
+        }
+        #endregion
+
+        public IList<PimsCompensationRequisition> GetAllByAcquisitionFileId(long acquisitionFileId)
+        {
+            return Context.PimsCompensationRequisitions
+                .Include(c => c.PimsCompReqH120s)
+                .AsNoTracking()
+                .Where(c => c.AcquisitionFileId == acquisitionFileId).ToList();
+        }
+
+        public PimsCompensationRequisition Add(PimsCompensationRequisition compensationRequisition)
+        {
+            User.ThrowIfNotAuthorized(Permissions.CompensationRequisitionAdd);
+            Context.PimsCompensationRequisitions.Add(compensationRequisition);
+
+            return compensationRequisition;
+        }
+
+        public void Delete(long compensationRequisitionId)
```

Comment:
This needs to change, either it should throw an error if it can't find an entity to delete or it should return false. If it returns false this method should be updated to the Try* syntax (ie: TryDelete). We use Try to indicate that it is ok for the method to not throw an exception if it is unable to complete.

Reply:
Shouldn't the Delete endpoint be idempotent?  I am removing this method since there was already a TryDelete method in the service.